### ISSUE TITLE
TGUI Bug when opening an "Exosuit Fabricator" on a private server.

### ISSUE THREAD

**Issue by LordTyrannus** [tgstation/tgstation#84681]
I was playing on my private tgstation server. When i tried to access the Exosuit Fabricator, a blue screen with text appears:

```
TypeError: Can't call method on undefined
   at Ce.exports (http://127.0.0.1:61643/tgui.bundle.js:410:63675)
   at Ce.exports (http://127.0.0.1:61643/tgui.bundle.js:410:70649)
   at Anonymous function (http://127.0.0.1:61643/tgui.bundle.js:410:58091)
   at values (http://127.0.0.1:61643/tgui.bundle.js:410:114195)
   at re (http://127.0.0.1:61643/tgui.bundle.js:299:254735)
   at cc (http://127.0.0.1:61643/tgui.bundle.js:9:69143)
   at Fd (http://127.0.0.1:61643/tgui.bundle.js:9:126671)
   at kd (http://127.0.0.1:61643/tgui.bundle.js:9:122014)
   at $f (http://127.0.0.1:61643/tgui.bundle.js:9:121947)
   at jc (http://127.0.0.1:61643/tgui.bundle.js:9:121795)
User Agent: Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.2; WOW64; Trident/7.0; .NET4.0C; .NET4.0E)
State: {"ckey":"myusername","interface":"ExosuitFabricator","window":{"key":"[0x2012f87]-main","size":null,"fancy":1,"locked":0}}

```

I tried refresh UI buttons and everything, didn't work, refreshing UI makes the blue bug screen white and uninteractable.

**Comment by Iamgoofball** [comment]
did you build tgui?

**Comment by LordTyrannus** [comment]
> did you build tgui?

Well i didn't specifically build TGUI but i compiled the codebase using the "build.bat". Do i need to build TGUI separately?

**Comment by TheVekter** [comment]
> > did you build tgui?
> 
> Well i didn't specifically build TGUI but i compiled the codebase using the "build.cmd". Do i need to build TGUI separately?

Try it and see if it fixes it, if not let me know.

**Comment by LordTyrannus** [comment]
Tried with a fresh install. tgui and tgstation dm codes compiled successfully (as far as console output goes). Still has the bug in game:

![ss13_bug_report](https://github.com/tgstation/tgstation/assets/37847853/8fbab66b-6d21-4269-8b36-5e07be469d23)

And after refreshing TGUI it becomes an empty ui element:

![ss13_bug_report2](https://github.com/tgstation/tgstation/assets/37847853/fe4cbd4d-e0c8-4ed6-850d-a4443cf5469b)

Some people on the internet mentioned that upgrading byond could solve this issue but my BYOND client is the latest one available. Could that be the case?

Also are there any kind of compilation logs or bug logs in the folders that i can share here?

**Comment by TheVekter** [comment]
I'm able to reproduce this on the latest BYOND version (515.1641).

```
 B7 in TGUI. The current application will be terminated. Send the copy of the following stack trace to an authorized Nanotrasen incident handler at https://github.com/tgstation/tgstation. Thank you for your cooperation. 
 TypeError: Can't call method on undefined
   at Ce.exports (http://127.0.0.1:62229/tgui.bundle.js:410:63675)
   at Ce.exports (http://127.0.0.1:62229/tgui.bundle.js:410:70649)
   at Anonymous function (http://127.0.0.1:62229/tgui.bundle.js:410:58091)
   at values (http://127.0.0.1:62229/tgui.bundle.js:410:114195)
   at re (http://127.0.0.1:62229/tgui.bundle.js:299:254735)
   at cc (http://127.0.0.1:62229/tgui.bundle.js:9:69143)
   at Fd (http://127.0.0.1:62229/tgui.bundle.js:9:126671)
   at kd (http://127.0.0.1:62229/tgui.bundle.js:9:122014)
   at $f (http://127.0.0.1:62229/tgui.bundle.js:9:121947)
   at jc (http://127.0.0.1:62229/tgui.bundle.js:9:121795)
User Agent: Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.2; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; .NET CLR 2.0.50727; .NET CLR 3.0.30729; .NET CLR 3.5.30729)
State: {"ckey":"vekter","interface":"ExosuitFabricator","window":{"key":"[0x200866e]-main","size":null,"fancy":1,"locked":0}}
```

**Comment by jlsnow301** [comment]
sounds like it's trying to iterate on a null value. ill take a look into it today

**Comment by LemonInTheDark** [comment]
closed by #84648
